### ISSUE TITLE
Command for delayed instructions untill passing a fix

### ISSUE THREAD

**Issue by erikquinn** [openscope/openscope#11]
Very similar to #10, this involves giving an instruction that the pilot is expected to execute *AFTER* passing a specified position (whether a fix, a distance from a fix, etc).

>"KAP515, depart BOWAN heading 360, vector visual approach runway 28"

In this example, the aircraft should continue direct to BOWAN, and then fly heading 360. You can also apply altiudes and speeds:

>"KAP515, depart BOWAN heading 360, then descend and maintain 3000 and reduce speed to 230, vector visual approach runway 28"

Originally reported in zlsa/atc#315.

**Comment by j-651** [comment]
@erikquinn in the first example, they don't necessarily need to be direct BOWAN for that instruction, do they? They could be 1-2 fixes and that would still be accurate, right? The direct would be a separate command, not baked in?

**Comment by erikquinn** [comment]
Correct, as long as BOWAN is on their route and they haven't already passed it, then they should be able to do it. Otherwise, they should respond unable because "BOWAN is not on our route" or some helpful response like that. Giving this instruction should _not_ mean direct, or any change to the route.

The way we would do this is to add a vector waypoint after BOWAN. Maybe getting the waypoint after BOWAN (if there is one)-- I'll call it FIXXB, and then doing a:
```
const routingToAdd = new RouteModel('BOWAN..#360..FIXXB');
fms.absorbRouteModel(routingToAdd);`
```

It could very well be as simple as this, and updating/adding the requisite tests. 🤔

**Comment by j-651** [comment]
It appears that this can be done through the route command, using the waypoint you want them to depart at followed by the heading. This seems to keep all other fixes prior that first waypoint as no direct is given.

Ex: `AAL123 route bowan..#360`

Was the desire to have an extra command designated solely for this or has `route`'s logic evolved since this issue was created?

**Comment by erikquinn** [comment]
Correct, we basically want a new command that has the exact same effect as the user entering `AAL123 route bowan..#360`.

**Comment by CanisLupus518** [comment]
@erikquinn as this is related to #10 would it be enough to extend the current "cross" command to allow a heading argument as well?  for instance "callsign, cross BOWAN S190 A60 H130"  (not sure if you want to use "H" for heading or use "#" is the prefix). Or are you looking for a new command\alias for "depart"?

**Comment by erikquinn** [comment]
Just contextually because of the real phraseology, it should be a different command. And it behaves slightly differently than cross anyway.

I think `at`, `depart`, `dep`, `leave`, `after` would cover everything. Perhaps "after" is a good one for the proper command name. (Unless it's a protected word? @n8rzz)

**Comment by erikquinn** [comment]
And to elaborate on its different behavior, it should really be not a "A50 S250" type thing, but rather reading normal commands which appear after the "after" fix, and it will execute those commands when the fms runs .moveToNextFix().

For example, `d 240 after yazuu x zizzi 190 at zizzi d 180 spd 250`-- meaning you could chain some whacky shit together like that:
- descend now to FL240
- after passing YAZUU, cross ZIZZI @ FL190
- after passing ZIZZI, descend to FL180 and maintain 250 knots

**Comment by n8rzz** [comment]
> Just contextually because of the real phraseology, it should be a different command. And it behaves slightly differently than cross anyway.
> 
> I think `at`, `depart`, `dep`, `leave`, `after` would cover everything. Perhaps "after" is a good one for the proper command name. (Unless it's a protected word? @n8rzz)

only in css.  so it's free to use here 😉

**Comment by erikquinn** [comment]
> > Just contextually because of the real phraseology, it should be a different command. And it behaves slightly differently than cross anyway.
> > I think `at`, `depart`, `dep`, `leave`, `after` would cover everything. Perhaps "after" is a good one for the proper command name. (Unless it's a protected word? @n8rzz)
> 
> only in css. so it's free to use here 😉

Ah, that's it!!! ::after 😂 

Glad I'm not crazy :)
